### PR TITLE
Form Submission Status on My VA

### PR DESCRIPTION
**Pull request by tpharrison**
## Summary
Solution for displaying the statuses of a veteran's form submissions on the My VA dashboard. 

High-level steps required to retrieve the form submission statuses for a user:
```mermaid
sequenceDiagram
    VA.gov Website->>+VA.gov API: Get Form Statuses
    VA.gov API->>+VA.gov API: Get User Account ID
    VA.gov API->>+VA.gov Database: Query form_submissions by Account ID
    VA.gov Database-->>-VA.gov API: List of benefits_intake_uuid's    
    VA.gov API->>+Lighthouse Benefits Intake API: POST /uploads/report
    Lighthouse Benefits Intake API-->>-VA.gov API: 200 OK: List of Form Statuses
    VA.gov API->>+VA.gov API: Parse and format results
    VA.gov API-->>-VA.gov Website: Return submission statuses
```

Responsibilities of core classes:
- `Forms::Submissions::Report`
  - Orchestrates the process of fetching and formatting submission statuses for a user account. 
- `Forms::Submissions::Gateway`
  - Fetches records from the `form_submissions` table and then retrieves their statuses from the Lighthouse Benefits Intake API.  Data is stored in `Forms::Submissions:Dataset`.
- `Forms::Submissions::Formatter`
  - Merges `forms_submissions` records with Benefits Intake status records and returns an array records.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/85942
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/85946

## Testing done
Added specs:
- `submission_statuses_request_spec.rb`
- `report_spec.rb`
- `formatter_spec.rb`

### CODE REVIEW

File: app/controllers/v0/my_va/submission_statuses_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+require 'forms/submission_statuses/report'
+
+module V0
+  module MyVA
+    class SubmissionStatusesController < ApplicationController
+      service_tag 'form-submission-statuses'
+      before_action :controller_enabled?
+      before_action { authorize :lighthouse, :access? }
+
+      def show
```
Comment (Thrillberg) [inline app/controllers/v0/my_va/submission_statuses_controller.rb:12]: This is a super clean, super idiomatic method and I love it! 😍

File: app/serializers/submission_status_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class SubmissionStatusSerializer
+  include JSONAPI::Serializer
+
+  attributes :id, :form_type, :status, :created_at, :updated_at
```
Comment (Thrillberg) [inline app/serializers/submission_status_serializer.rb:6]: Looks like the API can return a `message` and `detail` as well. Are we intentionally omitting those? Seems prudent to omit them, but just wanted to confirm. 🙂

File: config/routes.rb [hunk 3]
```diff
@@ -288,6 +288,10 @@
 
     resources :gi_bill_feedbacks, only: %i[create show]
 
+    namespace :my_va do
```
Comment (Thrillberg) [inline config/routes.rb:291]: Wow, our first `my_va` route? Nice! 🎉

File: lib/forms/submission_statuses/gateway.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        response = benefits_intake_service.bulk_status(uuids:)
+        raise response.body unless response.success?
+
+        response.body['data']
+      end
+
+      private
+
+      def benefits_intake_service
+        @service ||= BenefitsIntake::Service.new
```
Comment (Thrillberg) [inline lib/forms/submission_statuses/gateway.rb:38]: I think this service class is relatively new--very nice to see it used!

**Review by Thrillberg — APPROVED** [review]
Hey @tpharrison, I _really_ enjoyed reviewing this PR! It's very clearly written and I love the abstractions you've landed on. I'm pretty neutral on whether you want to omit the `fetch_` prefixes. They didn't strike me as an eyesore whatsoever. And for the use of `OpenStruct`, I also don't mind them being there as-is. It only shows up once in the implementation code, if I'm not mistaken, so it truly does seem like a one-off object. But I could see it going either way! Anyway, yeah, loved reviewing this.

File: app/serializers/submission_status_serializer.rb (hunk 2 above)
Comment (tpharrison) [reply app/serializers/submission_status_serializer.rb:6]: I'll add these two fields.

Comment (tpharrison) [reply app/serializers/submission_status_serializer.rb:6]: Added `message` and `detail`.  Thanks!

File: config/routes.rb (hunk 3 above)
Comment (tpharrison) [reply config/routes.rb:291]: 🎉

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**